emrun.py

```diff
@@ -400,6 +420,35 @@ def kill_browser_process():
     # Clear the process name to represent that the browser is now dead.
     processname_killed_atexit = ''
 
+  delete_emrun_safe_firefox_profile()
+
+
+# Heuristic that attempts to search for the browser process IDs that emrun spawned.
+# This depends on the assumption that no other browser process IDs have been spawned
+# during the short time perioed between the time that emrun started, and the browser
+# process navigated to the page.
+# This heuristic is needed because all modern browsers are multiprocess systems -
+# starting a browser process from command line generally launches just a "stub" spawner
+# process that immediately exits.
+def detect_browser_processes():
+  global previous_browser_processes, current_browser_processes, navigation_has_occurred
```

Comment:
Drop previous_browser_processes from this list since its not assigned to.

In fact, how about returning the list from this function, then so it has no effect on global state and instead do `current_browser_processes = detect_browser_processes` at the call site?

Reply:
Adjusted the `global` - I think I'd rather not change to that style, since then the call site will need to declare the `global`.